subprojects/core/src/main/java/org/gradle/process/internal/ExecFactory.java

```diff
@@ -57,6 +58,8 @@ interface Builder {
 
         Builder withExternalProcessStartedListener(ExternalProcessStartedListener externalProcessStartedListener);
 
+        Builder withExecHandleFactory(ClientExecHandleFactory clientExecHandleFactory);
+
```

Comment:
❓ Does it make sense to pass the `ClientExecHandleFactory` via `withExecHandleFactory()` (=can it ever be valid to not pass a `ClientExecHandleFactory`?) or shall we pass it via `withContext(ClientExecHandleFactory)`?

Reply:
We want to have specialized `ClientExecHandleFactory` in some lower scopes, so it make sense to `withExecHandleFactory`. Technically there is no difference from other services that are used in `ExecFactory` so to me it makes sense that it follows the same pattern